sonar-db/src/main/java/org/sonar/db/user/AuthorizationDao.java

```diff
@@ -129,4 +137,21 @@ public boolean isAuthorizedComponentKey(String componentKey, @Nullable Integer u
       MyBatis.closeQuietly(session);
     }
   }
+
+  private static class SelectUsersByPermissionAndProject implements Function<List<Long>, List<Long>> {
+    private final AuthorizationMapper mapper;
+    private final String role;
+    private final long componentId;
+
+    private SelectUsersByPermissionAndProject(AuthorizationMapper mapper, String role, long componentId) {
+      this.mapper = mapper;
+      this.role = role;
+      this.componentId = componentId;
+    }
+
+    @Override
+    public List<Long> apply(@Nonnull List<Long> partitionOfIds) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this "Nonnull" annotation to honor the overridden method's contract. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)